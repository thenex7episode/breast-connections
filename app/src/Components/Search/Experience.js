import React, { Component } from 'react';
import { Rate, Avatar, Icon } from 'antd';
import axios from 'axios';


export default class Experience extends Component {
    constructor(){
        super();
        this.state = {
            username: '',
            userImage: '',
            loggedUser: '',
            editMode: false,
            edit: false
        }
    }

    // return 'no Content' if nothing is available
    componentDidMount(){
        axios.get(`/api/posts/${this.props.user_id}`).then(data => {
            console.log(data)
            this.setState({username: data.data[0].username, userImage: data.data[0].imageurl})
        })
        axios.get(`/api/check-session/`).then( data => {
            console.log('session res', data)
            this.setState({loggedUser: data.data.username})
        })
    }

    render() {
        const { body, user_id, rating, deleteExperienceFn, experience_id } = this.props;
        return (
            <div style={{display: 'flex', justifyContent: 'space-between', borderBottom: '1px solid grey'}}>
                <div>
                    <Avatar alt='avatar' src={this.state.userImage} />
                    <div>{this.state.username}</div>
                </div>
                <div>
                    <Icon onClick={this.state.loggedUser === this.state.username ? () => this.setState({edit: !this.state.edit}): ''} style={{display: 'block',opacity: this.state.loggedUser === this.state.username ? '1' : '0', paddingBottom: '1em'}}type="ellipsis" />
                    <ul style={{display: this.state.edit ? 'block' : 'none'}} className='editContainer'>
                        <li onClick={() => deleteExperienceFn(experience_id)}><Icon type="delete" /> Delete</li>
                    </ul>
                    <p>{body}</p>
                    <Rate defaultValue={rating} disabled />
                </div>
            </div>
        );
    }
}