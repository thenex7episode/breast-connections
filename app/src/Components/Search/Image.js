import React, { Component } from 'react';
import axios from 'axios';

export default class Image extends Component {
    constructor(){
        super();
        this.state = {
            image: ''
        }
    }

    componentDidMount(){
        console.log('--render Photos in did mount', this.props.reference.photos || 'no photos')
        if(this.props.reference.photos){
            axios.post(`/api/googleimage/${this.props.reference.photos[0].photo_reference}/`).then(data => {
                console.log('ref', data.data)
                this.setState({image: data.data})
            })
        } else {
                this.setState({image: 'https://app.widiz.com/plugins1/support/public/files/products/no-img.png'})
        }
    }
    // componentWillReceiveProps(){
    //     this.props.reference.photos ? console.log('new photos', this.props.reference.photos) : null
    //     if(this.props.reference.photos){
    //         axios.post(`/api/googleimage/${this.props.reference.photos[0].photo_reference}/`).then(data => {
    //             console.log('ref', data.data)
    //             this.setState({image: data.data})
    //         })
    //     } else {
    //             this.setState({image: 'https://app.widiz.com/plugins1/support/public/files/products/no-img.png'})
    //     }
    // }

    render() {
        return (
            <div>
                <img maxwidth='400' maxheight='400' src={this.state.image} alt='resultphoto' />
            </div>
        );
    }
    
}