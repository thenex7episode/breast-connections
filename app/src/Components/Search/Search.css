.place_container {
    display: flex;
    justify-content: space-between;
    margin: 4em 2em;
    text-align: left;
}