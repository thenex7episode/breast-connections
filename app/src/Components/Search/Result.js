import React, { Component } from 'react';
import { Button, Spin, Icon } from 'antd';
import Image from './Image';
import axios from 'axios';
import Experience from './Experience';

const antIcon = <Icon type="loading" style={{ fontSize: 24 }} spin />;


export default class Result extends Component {
    constructor(){
        super();
        this.state = {
            exp: false,
            experiences: [],
            loading: true
        }
        this.deleteExperience = this.deleteExperience.bind(this);
    }

    getExperiences(id){
        axios.get(`/api/experiences/${id}`).then(data => {
            this.setState({experiences: data.data, loading: false})
        })
    }

    deleteExperience(id){
        axios.delete(`/api/experience/${id}/${this.props.place_id}`).then(data => {
            this.setState({experiences: data.data})
        })
    }

    render() {
        const { name, adress, gRating, reference, createMode, place_id } = this.props;
        const experienceList = this.state.experiences.map((el,i) => {
            return <Experience deleteExperienceFn={this.deleteExperience} experience_id={el.experience_id} key={i} rating={el.rating} body={el.body} user_id={el.user_id}/> 
        })
        // console.log('--Result', reference.photos || 'no photo')
        return (
            <div style={{textAlign: 'center'}}>
                <div key={name} className='place_container'>
                        <div>
                            <h6>{name}</h6>
                            <p>{adress}</p>
                            <div className='ratingContainer'>
                                <div>Google {gRating}</div>
                                <div>BC {gRating}</div>
                            </div>
                            <Button onClick={() => createMode(name, place_id)}>Share your Experience</Button> 
                        </div>
                        <Image width={272} reference={reference} />
                </div>
                <Button onClick={() => {
                    this.setState({exp: !this.state.exp})
                    this.getExperiences(place_id)
                }}>Experiences</Button> 
                <div style={{display: this.state.exp ? 'block' : 'none'}}>
                    {this.state.loading
                    ? <Spin indicator={antIcon} />
                    : <div>{experienceList}</div>}
                </div>
            </div>
        );
    }
}