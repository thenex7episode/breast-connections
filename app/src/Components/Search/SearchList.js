import React, { Component } from 'react';
import { List, Avatar, Icon, Button, Modal, Input, Rate } from 'antd';
import axios from 'axios';
import Image from './Image';
import Result from './Result';

const TextArea = Input.TextArea;

const IconText = ({ type, text }) => (
    <span>
      <Icon type={type} style={{ marginRight: 8 }} />
      {text}
    </span>
  );

export default class SearchList extends Component {
    constructor(props){
        super(props);
        this.state = {
            results: [],
            images: [],
            add: false,
            expInput: '',
            expRating: 0
        }
        this.createMode = this.createMode.bind(this)
    }

    getImage(ref){
        console.log(ref)
        if(ref.photos){
            axios.post(`/api/googleimage/${ref.photos[0].photo_reference}/`).then(data => {
                console.log('ref', data.data)
                return 'https://lh3.googleusercontent.com/p/AF1QipOSztHdWV4-PZf8hPegi8K3BCSuut9eToblAIsb=s1600-w400-h400'
            })
        } else {
            console.log('static image')
                return 'https://app.widiz.com/plugins1/support/public/files/products/no-img.png'
        }
    }

    addExperience(){

    }

    createMode(name){
        this.setState({add: true, currentPlace: name})
    }


    render() {
        // console.log('--render Searchlist!')
        const resultsList = this.props.results.map((el,i) => {
            const star = <Icon type="star" />
            let gRating = [];
            let bRating = [];
            for(let i = 0; i < el.rating; i++){
                gRating.push(star)
            }
            for(let i = 0; i < el.rating; i++){
                bRating.push(star)
            }
            return (
                <Result key={i} name={el.name} adress={el.adress} gRating={el.rating ? gRating : 'no rating'} reference={el} createMode={this.createMode}/> 
                // <li className='place_container'>
                //     <div>
                //         <h6>{el.name}</h6>
                //         <p>{el.adress}</p>
                //         <div className='ratingContainer'>
                //             <div>Google {el.rating ? gRating : 'no rating'}</div>
                //             <div>BC {el.rating ? bRating : 'no rating'}</div>
                //         </div>
                //         <Button onClick={() => this.setState({add: true, currentPlace: el.name})}>Share your Experience</Button> 
                //     </div>
                //     <Image width={272} reference={el} />
                // </li>
            )
        })
                console.log('--render Searchlist!', resultsList)
        return (
                    <div>
                        <div>{resultsList}</div>
                        <Modal title={`Create Experience for ${this.state.currentPlace}`} visible={this.state.add} onOk={() => this.addExperience()} onCancel={() => this.setState({add: false})}>
                            <TextArea onChange={e => this.setState({bodyInput: e.target.value})} rows={4} placeholder='what do you think?'/>
                            <Rate onChange={value => this.setState({expRating: value})}/>
                        </Modal>
                    </div>
        );
    }
}