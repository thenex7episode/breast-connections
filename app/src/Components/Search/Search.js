import React, { Component } from 'react';
import axios from 'axios';
import { Button, Select, Input, Slider } from 'antd';
import SearchList from './SearchList';
import './Search.css';
const Option = Select.Option;
const InputGroup = Input.Group;



export default class Search extends Component {
    constructor(){
        super()
        this.state = {
            results: [],
            selectedType: 'doctor',
            selectedLocation: '',
            locationResponse: '',
            timer: null,
            radius: 30
        }
    }

    isTyping(e){
        clearTimeout(this.state.timer)
        this.setState({timer: setTimeout(() => {
            console.log('run get Location')
            this.getLocation()
        }, 500)})
    }

    getLocation(){
        axios.get(`https://maps.googleapis.com/maps/api/geocode/json?address=${this.state.selectedLocation}&key=AIzaSyDP2xc5L8pWjHE2vgmIRDCK-834Q2eGA0A`).then(data => {
            console.log(data.data.results[0].geometry.location)
            this.setState({locationResponse: data.data.results[0].geometry.location})
        })
    }

    getGoogleResults(){
        console.log('results call')
        const type = {
            tags: this.state.selectedType,
            cords: this.state.locationResponse,
            radius: this.state.radius*1000
        }
        axios.post('/api/googlesearch/', type).then(data => {
            this.setState({results: data.data.results})
            console.log(this.state.results)
        })
    }

    render() {
        // console.log('--render Search!')
        const { selectedTags } = this.state;
        return (
            <div className='searchContainer' style={{padding: '5em', textAlign: 'center'}}>
                <h1>Search for Doctors and see rankings by BC Users</h1>
                <InputGroup style={{margin: '0 auto'}}compact>
                    <Select defaultValue="doctor" style={{ width: 120 }} onChange={(e) => this.setState({selectedType: e})}>
                        <Option value="doctor">Doctors</Option>
                        <Option value="hospital">Hospitals</Option>
                        <Option value="pharmacy">Pharmacies</Option>
                        <Option value="insurance_agency">Insurance Agencies</Option>
                    </Select>
                    <Input style={{width: '50%'}} value={this.state.selectedLocation} onChange={e => {
                        this.setState({selectedLocation: e.target.value})
                        this.isTyping()
                    }}/>
                    <Button>Search</Button>
                </InputGroup>
                <div style={{display: 'flex', justifyContent: 'space-between', width: '50%', margin: '0 auto', padding: '1em', alignItems: 'center'}}>
                    <div>Search Radius</div>
                    <Slider style={{margin: '1em auto', width: '50%', display: 'inline-block'}} tipFormatter={v => `${v} km`} onChange={e => this.setState({radius: e})} min={10} max={50} defaultValue={30} disabled={false} />
                    <div>{this.state.radius}</div>
                </div>
                <SearchList results={this.state.results}/>
            </div>
        );
    }
}