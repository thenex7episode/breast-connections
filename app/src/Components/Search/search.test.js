import Search from './Search'

it("renders correctly", () => {
    const wrapper = shallow(
        <Search />
    );
    expect(wrapper).toMatchSnapshot()
});

it("renders correctly", () => {
    const wrapper = mount(
        <Search />
    );
    const text = wrapper.find('h1').first().text()
    expect(text).toEqual('Search for Doctors and see rankings by BC Users')
});