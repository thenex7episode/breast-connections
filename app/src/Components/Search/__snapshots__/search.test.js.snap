// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "padding": "5em",
    }
  }
>
  <h1>
    Search for Doctors and see rankings by BC Users
  </h1>
  <div
    className="inputContainer"
  >
    <Input
      disabled={false}
      onChange={[Function]}
      prefixCls="ant-input"
      type="text"
      value=""
    />
    <Select
      choiceTransitionName="zoom"
      defaultValue="doctor"
      onChange={[Function]}
      prefixCls="ant-select"
      showSearch={false}
      style={
        Object {
          "width": 120,
        }
      }
      transitionName="slide-up"
    >
      <Option
        value="doctor"
      >
        Doctors
      </Option>
      <Option
        value="hospital"
      >
        Hospitals
      </Option>
      <Option
        value="pharmacy"
      >
        Pharmacies
      </Option>
      <Option
        value="insurance_agency"
      >
        Insurance Agencies
      </Option>
    </Select>
    <Button
      ghost={false}
      loading={false}
      onClick={[Function]}
      prefixCls="ant-btn"
    >
      Get Results
    </Button>
  </div>
  <SearchList
    results={Array []}
  />
</div>
`;
